sensor_test.py

```diff
@@ -0,0 +1,240 @@
+    4. Get sensor reading.
+    5. Wait for interval specified in Step 1.
+    6. Get sensor reading.
+    7. Validate that reading from Step 4 doesn't equal reading from Step 6.
+    """
+    print("Set sensor reading interval to 1")
+    new_reading_interval = 1
+    set_sensor_reading_interval(new_reading_interval)
+
+    print("Get sensor info")
+    updated_sensor_info = get_sensor_info()
+
+    print("Validate that sensor reading interval is set to interval from Step 1")
+    current_reading_interval = updated_sensor_info.get("reading_interval")
+    assert (
+        current_reading_interval == new_reading_interval
+    ), f"Sensor reading interval was not updated, expected interval {new_reading_interval}"
+
+    print("Get sensor reading")
+    sensor_reading_before_wait = get_sensor_reading()
+
+    print("Wait for interval specified in Step 1")
+    sleep(new_reading_interval)
+
+    print("Get sensor reading")
+    sensor_reading_after_wait = get_sensor_reading()
+
+    print("Validate that reading from tep 4 doesn't equal reading from Step 6")
+    assert (
+        sensor_reading_before_wait != sensor_reading_after_wait
+    ), f"Sensor readings after {new_reading_interval} second of waiting are equal to readings before waiting"
+
+
+def test_update_sensor_firmware(get_sensor_info, update_firmware):
+    """
+    1. Get original sensor firmware version.
+    2. Request firmware update.
+    3. Get current sensor firmware version.
+    4. Validate that current firmware version is +1 to original firmware version.
+    5. Repeat steps 1-4 until sensor is at max_firmware_version - 1.
+    6. Update sensor to max firmware version.
+    7. Validate that sensor is at max firmware version.
+    8. Request another firmware update.
+    9. Validate that sensor doesn't update and responds appropriately.
+    10. Validate that sensor firmware version doesn't change if it's at maximum value.
+    """
+    print("Get original sensor firmware version")
+    sensor_info = get_sensor_info()
+    original_firmware_version = sensor_info.get("firmware_version")
+
+    print("Request firmware update")
+    update_firmware()
+    wait(
+        func=get_sensor_info,
+        condition=lambda x: isinstance(x, dict),
+        tries=5,
+        timeout=3,
+    )
+
+    print("Get current sensor firmware version")
+    updated_sensor_info = get_sensor_info()
+    current_firmware_version = updated_sensor_info.get("firmware_version")
+
+    print("Validate that current firmware version is +1 to original firmware version")
+    assert (
+        current_firmware_version == original_firmware_version + 1
+    ), "Sensor firmware was not updated"
+
+    print("Repeat steps 1-4 until sensor is at max_firmware_version - 1")
+    max_firmware_version = 15
+    while True:
+        update_firmware()
+        wait(
+            func=get_sensor_info,
+            condition=lambda x: isinstance(x, dict),
+            tries=5,
+            timeout=3,
```

Comment:
It's good that you're trying to reduce pressure on the sensor by increasing wait interval. The only thing to consider is the tradeoff you're making in terms of time. Sensor could come online right after you've put the test to sleep for 3 seconds, so you'll get an "empty" wait. It's not a big deal for one test but would be a good five minutes for a hundred.

There are no hard and fast rules about this, so we're left with our best judgement. Modern systems are capable of handling much more than a request per second, so in most cases you're better off optimizing for a response as soon as possible rather than going easy on the system.